File: pyg_interface/src/main/java/cn/itcast/core/service/AddressService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.address.Address;

import java.util.List;

public interface AddressService {

    /**
     * 根据用户查询地址
     * @param userId
     * @return
     */
    public List<Address> findListByUserId(String userId );

    /**
     * 添加收件信息
     * @param address
     */
    void add(Address address);

    /**
     * 查询一个
     * @param id
     * @return
     */
    Address findOne(Long id);

    /**
     * 修改
     * @param address
     */
    void update(Address address);

    /**
     * 删除
     * @param id
     */
    void delete(Long id);
}

File: web_user/src/main/webapp/js/service/addressService.js
//服务层
app.service('addressService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../address/findAll.do');
	}

	//查询实体
	this.findById=function(id){
		return $http.get('../address/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../address/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../address/update.do',entity );
	}
	//通过id删除
	this.dele=function(id){
		return $http.get('../address/delete.do?id='+id);
	}
	//查询所有省份
	this.selectProvincesList=function () {
		return $http.get('../address/selectProvincesList.do?');
    }
    //查询省份下的所有城市列表
    this.findByprovinceid=function (provincesid) {
		return $http.get('../address/findByprovinceid.do?provincesid='+provincesid);
    }
    //查询城市下的所有区域列表：
    this.findBycityid=function (citiesid) {
        return $http.get('../address/findBycityid.do?citiesid='+citiesid);
    }

});
